src/common/url-parser.ts

```diff
@@ -10,6 +10,26 @@ export class UrlParser {
     public areURLHostNamesEqual(urlA: string, urlB: string): boolean {
         const urlAObj = new URL(urlA);
         const urlBObj = new URL(urlB);
-        return urlAObj.hostname === urlBObj.hostname;
+
+        if (this.areBothFileUrls(urlAObj, urlBObj)) {
+            return this.areFileUrlsEqual(urlA, urlB);
+        }
+
+        return this.areUrlProtocolsEqual(urlAObj, urlBObj) && urlAObj.hostname === urlBObj.hostname;
+    }
+
+    public areFileUrlsEqual(urlA: string, urlB: string): boolean {
+        const urlAObj = new URL(urlA);
+        const urlBObj = new URL(urlB);
+
+        return urlAObj.href === urlBObj.href;
+    }
+
+    private areBothFileUrls(urlAObj: URL, urlBObj: URL): boolean {
+        return this.areUrlProtocolsEqual(urlAObj, urlBObj) && urlAObj.protocol === 'file:' && urlBObj.protocol === 'file:';
```

Comment:
We only need 2 of these 3 checks (if A=B and B=C then we know A=C).

Reply:
Yes. I added the `A=B` check later to make it more tight but forgot to remove `A=C` check. Will do.